### ISSUE TITLE
[Windows][Firefox] Encoded icons of Mozilla's default bookmarks in exported file have wrong MIME type

### ISSUE THREAD

**Issue by e-d-n-a** [FlyingWolFox/Bookmarks-to-Dir#1]
Hey,

thank you for making this code and especially the package ([Netscape-Bookmarks-File-Parser](https://github.com/FlyingWolFox/Netscape-Bookmarks-File-Parser)) publicly available,
it was just what I needed and it works great, for the most part. :)

**Here is Issue-Nr.1 of the 2, that I got for you:**

In Firefox there are 4 default bookmarks in the "Bookmarks Menu" of "All Bookmarks".
Two of them have the Firefox icon and two have the Mozilla logo.

_Links with Mozilla logo:_
```
https://www.mozilla.org/de/contribute/
https://www.mozilla.org/de/about/
```

The issue is, that **these icons are actually encoded svg drawings**, but still appear as base64-encoded pngs in an exported bookmarks.html.

_Exported as this:_
`data:image/png;base64,PHN2ZyB4bWxucz0naHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmcnIHdpZHRoPScxNicgaGVpZ2h0PScxNic+IDxwYXRoIGQ9J00wIDBoMTZ2MTZIMHonLz4gPHBhdGggZD0nTTEzLjk5NCAxMC4zNTZIMTVWMTJoLTMuMTcxVjcuNzQxYzAtMS4zMDgtLjQzNS0xLjgxLTEuMjktMS44MS0xLjA0IDAtMS40Ni43MzctMS40NiAxLjh2Mi42M2gxLjAwNlYxMkg2LjkxOFY3Ljc0MWMwLTEuMzA4LS40MzUtMS44MS0xLjI5MS0xLjgxLTEuMDM5IDAtMS40NTkuNzM3LTEuNDU5IDEuOHYyLjYzaDEuNDQxVjEySDF2LTEuNjQ0aDEuMDA2VjYuMDc5SDFWNC40MzVoMy4xNjh2MS4xMzlhMi41MDcgMi41MDcgMCAwIDEgMi4zLTEuMjlBMi40NTIgMi40NTIgMCAwIDEgOC45MzEgNS45MSAyLjUzNSAyLjUzNSAwIDAgMSAxMS40IDQuMjg0IDIuNDQ4IDIuNDQ4IDAgMCAxIDE0IDYuOXYzLjQ1OHonIGZpbGw9JyNmZmYnLz4gPC9zdmc+`

This appears to be a fault on behalf of Mozilla during bookmark export (using the wrong MIME-type),
but it does lead to an error while running your code and trying to decode it as an png to save it on the filesystem as an icon-file.

Maybe you want to extend the checking after decoding the base64-data and before handing it to PIL.

**See this [problematic line in the code](https://github.com/FlyingWolFox/Bookmarks-to-Dir/blob/01670e3a1da5d6b7bf673d2a3471f194ad4c36d0/bookmark_to_dir.py#L36).**

I fixed it easily myself by manually correcting the MIME-type for those two links in the html file, but it would be nice, if they hadn't caused this issue!

**Sorry, for not providing a PR!**

**Comment by FlyingWolFox** [comment]
Interesting, these bookmarks have normal png icons when saved naturally, but these default bookmarks aren't. Anyway, it can check if the base64 data is a png file by checking the presence of the png header, it has to start with "iVBORw0KGgo", if it doesn't match the favicon is ignored.

However, svg favicons are a thing and this tool doesn't support it. I'm going to add it, probably with cairo.

However having the wrong MIME type is kinda bad, I'm thinking of filliling a bug report or something. Gotta test if this also happens on Edge.

Good to know that this tool serves you, thank you for using :D

**Comment by FlyingWolFox** [comment]
Solved with #4 
AFAIK svg file icons aren't supported, so it converts the svg to a png. It'll verify MIME-type and file header so no errors due to wrong MIME-type.
